src/data/waffleFlags.test.js

```diff
@@ -0,0 +1,167 @@
+import { getConfig, mergeConfig } from '@edx/frontend-platform';
+
+import {
+  isWaffleFlagEnabled,
+  processUrlWaffleFlags,
+  WAFFLE_PREFIX,
+  waffleInterceptor,
+} from './waffleFlags';
+
+/* eslint-disable import/prefer-default-export */
+
+/**
+ * Takes a potentially Asyncronous function and invokes it wraped as a promise.
+ * @param {Function} fn
+ * @param {Function} then
+ * @returns {Promise<void>}
+ */
+const executePossiblyAsyncFunction = (fn, then) => Promise.resolve(fn()).then(then);
+
+describe('Testing the Testers: executePossiblyAsyncFunction', () => {
+  it('should execute as a Function', () => {
+    const fn = jest.fn();
+    const then = jest.fn();
+
+    executePossiblyAsyncFunction(fn, then)
+      .then(() => {
+        expect(fn).toHaveBeenCalledTimes(1);
+        expect(then).toHaveBeenCalledTimes(1);
+      });
+  });
+
+  it('should execute as a AsyncFunction', async (done) => {
+    const fnInnerMock = jest.fn();
+    const then = jest.fn();
+    const fn = async () => { fnInnerMock(); };
+
+    executePossiblyAsyncFunction(fn, then)
+      .then(() => {
+        expect(fnInnerMock).toHaveBeenCalledTimes(1);
+        expect(then).toHaveBeenCalledTimes(1);
+      })
+      .then(done);
+  });
+});
+
+/**
+ * Set our JSDOM Window and Document Location
+ *
+ * **NOTE**: You *MUST* include your tests async DoneCallback as the final action of your `perform`'s closure.
+ * failure to do so will lead to test timeouts.
```

Comment:
Question: Why is this comment located here, as opposed to the test files where `done()` is called?

Asking because placing this comment here makes it sound like there is something in this function that requires a done callback, but I'm not sure what it is.

My understanding was that the done callback pattern is only necessary when we have a test with a callback we don't await. But what callback are we are not awaiting in our tests - could you point me to them? Or am I missing something?

Reply:
I have added to the documentation, feel free to open this again if its confusing.